cl/corpus_importer/tests.py

```diff
@@ -681,3 +682,240 @@ def test_short_opinion_matching(self) -> None:
 
         bad_match = compare_documents(harvard_characters, bad_characters)
         self.assertEqual(bad_match, 81)
+
+    def test_winnow_case_name(self) -> None:
+        """Test function winnow_case_name() to reduce case names to a set of words
+        Some names were wiped out when the name only contains abbreviations/acronyms
+        """
+
+        # (case name full, case name)
+        data_case_1 = (
+            "In the matter of S.J.S., a minor child. D.L.M. and D.E.M. "
+            "Petitioners/Respondents v. T.J.S.",
+            "D.L.M. v. T.J.S.",
+        )
+
+        self.assertEqual(
+            winnow_case_name(data_case_1[0]),
+            {
+                "respondents",
+                "minor",
+                "petitioners",
+                "sjs",
+                "tjs",
+                "dlm",
+                "dem",
+                "child",
+            },
+        )
+        self.assertEqual(
+            winnow_case_name(f"{data_case_1[0]} {data_case_1[1]}"),
+            {
+                "respondents",
+                "minor",
```

Comment:
this function seems particularly long.

Reply:
Can we group all the testing code into a list of tuples (or whatever) at the the start and loop thru it?

Reply:
I added some cases in json format, imported them in the test and execute the part of the harvard importer where it checks that the case names don't overlap